shell/browser/api/electron_api_base_window.cc

```diff
@@ -1071,9 +1079,12 @@ void BaseWindow::ResetBrowserViews() {
                            v8::Local<v8::Value>::New(isolate(), item.second),
                            &browser_view) &&
         !browser_view.IsEmpty()) {
-      if (browser_view->web_contents()) {
-        window_->RemoveBrowserView(browser_view->view());
+      // There's a chance that the BrowserView may have been reparented - only
+      // reset if the owner window is *this* window.
+      auto* owner_window = browser_view->web_contents()->owner_window();
+      if (browser_view->web_contents() && owner_window == window_.get()) {
         browser_view->web_contents()->SetOwnerWindow(nullptr);
+        owner_window->RemoveBrowserView(browser_view->view());
```

Comment:
Does the order of calls matter ? Does the view cleanup code path depend on having a valid browser window ?

i-e Removing view from window before unsetting the window pointer or vice-versa.

Reply:
it doesn't that I was able to determine through any testing - this is also identical to the codepath we take when removing a `BrowserView` directly via API:

https://github.com/electron/electron/blob/e4c386bbd51a510fb8096e272670c71f0765b843/shell%2Fbrowser%2Fapi%2Felectron_api_base_window.cc#L784-L787